src/schema.js

```diff
@@ -12,6 +12,37 @@ const IGNORE_DELTA_METHODS = [
 
 const SUCCESS_KEY = 'qSuccess';
 
+const hasOwnProperty = Object.prototype.hasOwnProperty;
+
+/**
+* Returns the camelCase counterpart of a symbol.
+* @param {String} symbol The symbol.
+* @return the camelCase counterpart.
+*/
+function toCamelCase(symbol) {
+  return symbol.substring(0, 1).toLowerCase() + symbol.substring(1);
+}
+
+/**
+ * A facade function that allows parameters to be passed either by name
+ * (through an object), or by position (through an array).
+ * @param {Function} base The function that is being overriden. Will be
+ *                        called with parameters in array-form.
+ * @param {Object} defaults Parameter list and it's default values.
+ * @param {*} params The parameters.
+ */
+function namedParamFacade(base, defaults, ...params) {
+  if (params && params.length === 1 && typeof params[0] === 'object') {
```

Comment:
No need to check for `params` truthyness, `...params`  produces an empty array if there are none.